### PR TITLE
Additional config options

### PR DESCRIPTION
**Pull request by mergwyn**
Add support for configuring:
	• user
	• group
	• port-forwarding-enabled
	• watch-dir

Allow absoltue directory names for *dir directives in config.

Tidy up handling of config file on ubuntu 16.04 where the homedir should be /home/user/config/transmission-daemon"

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -6,14 +6,17 @@
     'transmission-daemon',
   ]
 
-  if $facts['os']['release']['full'] == '16.04' {
+  #if $facts['os']['release']['full'] == '16.04' {
+  if ($::operatingsystem == 'Ubuntu' and versioncmp($::operatingsystemrelease, '16.04')) {
```
Comment (craigwatson) [inline manifests/params.pp]: This is the old Puppet3 way of using facts. Please update this to use the new `$facts` hash.

File: manifests/params.pp [hunk 2]
```diff
@@ -6,14 +6,17 @@
...
     $use_systemd = true
-    $home_dir    = '/home/debian-transmission'
+    $home_dir    = "/home/${::transmission::user}/.config/transmission-daemon"
+    $config_dir  = "${::transmission::params::home_dir}"
```
Comment (craigwatson) [inline manifests/params.pp]: Raw variables don't need to be quoted.

File: templates/systemd.erb [hunk 3]
```diff
@@ -8,9 +8,16 @@ Description=Transmission BitTorrent Daemon
 After=network.target
 
 [Service]
+<% if scope['::transmission::user'] -%>
+User=<%= scope['::transmission::user'] %>
+<% else -%>
 User=debian-transmission
```
Comment (craigwatson) [inline templates/systemd.erb]: This will need to be removed if you're setting the user dynamically above.

File: manifests/config.pp [hunk 4]
```diff
@@ -20,20 +20,25 @@
   # == Defaults
 
   File {
-    owner   => 'debian-transmission',
-    group   => 'debian-transmission',
+    owner   => "${::transmission::user}",
```
Comment (craigwatson) [inline manifests/config.pp]: Raw variables don't need to be quoted.

**Review by craigwatson — CHANGES_REQUESTED** [review]
Hi,

Thanks for the PR, there are a few issues which need to be fixed however.

I'd be more than happy to merge the PR once the comments have been addressed :)

Thanks,
Craig

**Comment by mergwyn** [comment]
Hi Craig, thanks for your comments.  I've made the changes you suggested (and worked out how to enable the facts has in puppet 3!).

**Comment by mergwyn** [comment]
Sorry - closed the PR by mistake.

**Comment by craigwatson** [comment]
This is being closed in favour of #3

**Pull request closed without merging**